Makefile

```diff
@@ -177,18 +195,36 @@ SFCFLAGS := --matching
 # We can't use the C preprocessor for this because it won't substitute inside string literals.
 BUILD_DIR_REPLACE := sed -e 's|$$(BUILD_DIR)|$(BUILD_DIR)|g'
 
-CFLAGS           += -G 0 -non_shared -Xcpluscomm -nostdinc -Wab,-r4300_mul
+GBI_DEFINES := -DF3DEX_GBI_2 -DF3DEX_GBI_PL -DGBI_DOWHILE
 
-WARNINGS         := -fullwarn -verbose -woff 624,649,838,712,516,513,596,564,594,807
-ASFLAGS          := -march=vr4300 -32 -G0
-GBI_DEFINES 	 := -DF3DEX_GBI_2 -DF3DEX_GBI_PL -DGBI_DOWHILE
-COMMON_DEFINES   := -D_MIPS_SZLONG=32 $(GBI_DEFINES)
-AS_DEFINES       := $(COMMON_DEFINES) -DMIPSEB -D_LANGUAGE_ASSEMBLY -D_ULTRA64
-C_DEFINES        := $(COMMON_DEFINES) -DLANGUAGE_C -D_LANGUAGE_C
-ENDIAN           := -EB
+ifeq ($(COMPILER),gcc)
+  # MIPS options
+  CFLAGS           += -G 0 -march=vr4300 -mtune=vr4300 -mfix4300 -mabi=32 -mno-abicalls -mdivide-breaks
+  # C dialect options
+  CFLAGS           += -nostdinc -fno-PIC -fno-common -ffreestanding -fbuiltin -fno-builtin-sinf -fno-builtin-cosf -funsigned-char
 
-OPTFLAGS := -O2 -g3
-MIPS_VERSION := -mips2
+  WARNINGS         := $(CC_CHECK_WARNINGS)
+  ASFLAGS          := -march=vr4300 -32 -G0
+  COMMON_DEFINES   := $(GBI_DEFINES)
+  AS_DEFINES       := $(COMMON_DEFINES) -DMIPSEB -D_LANGUAGE_ASSEMBLY -D_ULTRA64
+  C_DEFINES        := $(COMMON_DEFINES) -D_LANGUAGE_C
+  ENDIAN           :=
```

Comment:
Why omit this?
```suggestion
  ENDIAN           := -EB
```

Reply:
It's an architecture default, it doesn't really need to be explicit. We already rely on `-mhard-float` being implicit from the arch